devtron-labs/devtron
# feat: Audit logs for docker and git repositories

# Description

In order to save audit logs , new tables for history will be created. 
For saving history of git repository, Git Materials history table will be created.
For saving history of docker repository, Ci Template history table will be created.
For saving history of ci pipeline material and ci template override, ci pipeline history table will be created.



## How Has This Been Tested?

For testing Audit logs of git repositories:-

1. Add a git repository, then verify a row is created in git_material_history for creation.
2. Go to same git repo. and change the clone directory to diff. directory, update the git repo and verify creation of another record in the git_material_history.
3. After that, verify Pull Submodules Recursively updating creates another record in git_material_history or not. 
4. Change the same git material to diff. git repo name and check another row inserted in git_material_history or not
5. Delete that git repository, and check if this action introcuces a row in git_material_history or not."
6. Create multiple git repos. and do above actions to check if it works with multi-git

For testing Audit logs of ci template repository:-

1. Add a container registory, and see if it is recorded in the ci_template_history table.
2. Edit and change the container repository to something else and check if that is recorded in the ci_template_history table."
3. Try out with different options(creating, editing,deleting) for selecting dockerfile and check if that is recorded in the ci_template_history table.
4. Play around with Advance options, add, edit them and check if that is recorded in the ci_template_history table."

For testing audit logs of ci pipeline :-

1. Edit Build Pipeline fields and check if the changes are recorded in ci_pipeline_history table.
2. Check if changing the Manually and Automatically field makes a record in  ci_pipeline_history table.
3. Change scan for vulnerabilities option multiple times to see if changes are recorded in the ci_pipeline_history table.
4. Play with override option to see if clicks are recorded in the ci_pipeline_history table.

## Checklist:

* [ ] Does this PR requires documentation updates?
* [ ] I have tested it for all user roles.
* [ ] I have added all the required unit/api test cases.

## Does this PR introduce a user-facing change?

```release-note

```
